src/geometry/warp.jl

```diff
@@ -40,7 +40,7 @@ function (h::Homography{K})(x::SVector{3, L}) where {L, K}
 end
 
 function (h::Homography{K})(x::AbstractVector{L}) where {L, K}
-    x = SVector(x...)
+    x = SVector(x...)::SMatrix{3, 3, L, 9}
```

Comment:
Are you sure this ever works? Is this method covered by the test?

Reply:
Sorry for the goof-up, should not have been this. I have rectified it and added tests for the same.